onnx/backend/test/case/node/regex_full_match.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+import onnx
+from onnx.backend.test.case.base import Base
+from onnx.backend.test.case.node import expect
+
+
+class RegexFullMatch(Base):
```

Comment:
We should add a test with there is no match at all, a null dimension. Do we assume the regular expression is always right?

Reply:
Behaviour should be undefined if an invalid pattern is provided. A reasonable approach may be to raise (there seems to be precedent for this in [Squeeze](https://github.com/onnx/onnx/blob/main/docs/Operators.md#squeeze)) but I'd prefer to let implementations decide.

Reply:
It seems fine for the op spec to clarify that the pattern is expected to be valid. (Then, the behavior will be implementation-dependent.) For the reference-implementation, it seems a good idea to raise an exception with a clear message.